### ISSUE TITLE
ansible-lint exits when using newer role dependency dict structure

### ISSUE THREAD

**Issue by sjmiller** [willthames/ansible-lint#174]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible --version: 2.1.0.0
ansible-lint --version: 3.1.2
```

- ansible installation method: homebrew
- ansible-lint installation method: pip

# Desired Behaviour

Command should not exit with error when role dependency is in newer dictionary format

# Actual Behaviour (Bug report only)

When a role dependency in `meta/main.yml` is defined like so:
```
dependencies:
  - name: my-role
    src: https://example.com/my-role.git
    scm: git
```
Errors with with:
`role dict {u'src': u'https://example.com/my-role.git', u'name': u'my-role', u'scm': u'git'} does not contain a "role" key`

Defining a role dependency providing the rolespec format in the `role` key appears to be deprecated in ansible 2.1.0.0.

**Comment by azenk** [comment]
I'm seeing the same bug.  This is the ansible code that seems to be used for parsing such declarations.  https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/role/requirement.py#L122  It looks like the logic around parsing these is complicated.   Would it make sense to use the ansible RoleRequirement parser here if it's available?  Obviously that would only work if ansible 2.x were installed.

**Comment by willthames** [comment]
@azenk, `RoleRequirement` is used for the requirements.yml file when installing roles through ansible-galaxy. `RoleDefinition`'s `_load_role_name` is all we need here, and the current code just needs an equivalent to https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/role/definition.py#L116 to work.

**Comment by willthames** [comment]
Pushed test release as 3.3.0rc1

**Comment by willthames** [comment]
Fixed in released 3.3.0

**Comment by bbaassssiiee** [comment]
REGRESSION

Successfully installed MarkupSafe-1.0 ansible-2.3.0.0 ansible-lint-3.4.12 asn1crypto-0.22.0 cffi-1.10.0 cryptography-1.8.1 idna-2.5 jinja2-2.9.6 paramiko-2.1.2 pyasn1-0.2.3 setuptools-35.0.2
demo:buildserver bas$ ansible-lint provision.yml 
role dict {u'src': u'base_java8'} does not contain a "role" or "name" key
